File: TheaterSuggestions/Java/SeatingPlaceSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/SuggestionMade.java
package org.weaveit.seatingplacesuggestions;

import java.util.List;

public class SuggestionMade {

    private final List<SeatingPlace> suggestedSeats;
    private final int partyRequested;
    private final PricingCategory pricingCategory;

    public SuggestionMade(SeatingOptionSuggested seatingOptionSuggested) {
        this.suggestedSeats = seatingOptionSuggested.seats();
        this.partyRequested = seatingOptionSuggested.partyRequested();
        this.pricingCategory = seatingOptionSuggested.pricingCategory();
    }

    public List<String> seatNames() {
        return suggestedSeats.stream().map(SeatingPlace::toString).toList();
    }

    public boolean matchExpectation() {
        return suggestedSeats.size() == partyRequested;
    }

    public PricingCategory pricingCategory() {
        return pricingCategory;
    }
}

File: TheaterSuggestions/CSharp/SeatingPlaceSuggestions/SuggestionMade.cs
namespace SeatsSuggestions;

/// <summary>
///     Occurs when a Suggestion is made.
/// </summary>
public class SuggestionMade(SeatingOptionSuggested seatingOptionSuggested)
{
    private readonly List<SeatingPlace> _suggestedSeats = seatingOptionSuggested.Seats;

    public int PartyRequested { get; } = seatingOptionSuggested.PartyRequested;
    public PricingCategory PricingCategory { get; } = seatingOptionSuggested.PricingCategory;

    public IReadOnlyList<SeatingPlace> SuggestedSeats => _suggestedSeats;

    public IEnumerable<string> SeatNames()
    {
        return _suggestedSeats.Select(s => s.ToString());
    }

    public bool MatchExpectation()
    {
        return _suggestedSeats.Count == PartyRequested;
    }
}